packages/api-explorer/src/CodeSample.jsx

```diff
@@ -5,61 +5,133 @@ const Oas = require('./lib/Oas');
 
 const { Operation } = Oas;
 
-const generateCodeSnippet = require('./lib/generate-code-snippet');
+const syntaxHighlighter = require('@readme/syntax-highlighter');
 
-function CodeSample({ oas, setLanguage, operation, formData, language }) {
-  return (
-    <div className="code-sample tabber-parent">
-      {(() => {
-        if (!oas[extensions.SAMPLES_ENABLED]) {
-          return <div className="hub-no-code">No code samples available</div>;
-        }
+const generateCodeSnippet = require('./lib/generate-code-snippet');
 
-        const snippet = generateCodeSnippet(oas, operation, formData, language);
+class CodeSample extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      customCodeSampleTab: 0,
+    };
+    this.setCustomCodeSampleTab = this.setCustomCodeSampleTab.bind(this);
+  }
+  setCustomCodeSampleTab(index) {
```

Comment:
How come we need to do this? How come `setLanguage()` isnt enough?

Reply:
It's so that the code samples body can switch along with  whatever language is set here https://github.com/readmeio/api-explorer/pull/108/files#diff-c199d2e44031baaac9109fd82d5610cdR54

I can try to make the styling work with setLanguage?